tstest/tstest.go

```diff
@@ -6,12 +6,31 @@ package tstest
 
 import (
 	"context"
+	"testing"
 	"time"
 
 	"tailscale.com/logtail/backoff"
 	"tailscale.com/types/logger"
 )
 
+// Replace replaces the value of target with val.
+// It returns a function that restores target to its original value.
+// After test t completes, the restore function is called automatically.
+func Replace[T any](t *testing.T, target *T, val T) (restore func()) {
```

Comment:
You're not calling restore yet anywhere directly, are you planning to soon? If not I'd leave it out, slightly simpler function signature makes this all easier to read and reason about.

Reply:
good point, removed and added a test.